src/generic/gridsel.cpp

```diff
@@ -975,222 +572,330 @@ void wxGridSelection::ClearSelection()
-        {
-            if (numCols > 0)
-            {
-                // If rows inserted, increase row counter where necessary
-                coords.SetCol(col + numCols);
-            }
-            else if (numCols < 0)
-            {
-                // If rows deleted ...
-                if ((size_t)col >= pos - numCols)
-                {
-                    // ...either decrement row counter (if row still exists)...
-                    coords.SetCol(col + numCols);
-                }
-                else
-                {
-                    // ...or remove the attribute
-                    m_cellSelection.RemoveAt(n);
-                    n--;
-                    count--;
-                }
-            }
-        }
-    }
-
-    count = m_blockSelectionTopLeft.GetCount();
-    for ( n = 0; n < count; n++ )
-    {
-        wxGridCellCoords& coords1 = m_blockSelectionTopLeft[n];
-        wxGridCellCoords& coords2 = m_blockSelectionBottomRight[n];
-        wxCoord col1 = coords1.GetCol();
-        wxCoord col2 = coords2.GetCol();
+        wxGridBlockCoords& block = m_selection[n];
+        wxCoord col1 = block.GetLeftCol();
+        wxCoord col2 = block.GetRightCol();
 
         if ((size_t)col2 >= pos)
         {
             if (numCols > 0)
             {
                 // If rows inserted, increase row counter where necessary
-                coords2.SetCol(col2 + numCols);
+                block.SetRightCol(col2 + numCols);
                 if ((size_t)col1 >= pos)
-                    coords1.SetCol(col1 + numCols);
+                    block.SetLeftCol(col1 + numCols);
             }
             else if (numCols < 0)
             {
                 // If cols deleted ...
                 if ((size_t)col2 >= pos - numCols)
                 {
                     // ...either decrement col counter (if col still exists)...
-                    coords2.SetCol(col2 + numCols);
+                    block.SetRightCol(col2 + numCols);
                     if ( (size_t) col1 >= pos)
-                        coords1.SetCol( wxMax(col1 + numCols, (int)pos) );
+                        block.SetLeftCol( wxMax(col1 + numCols, (int)pos) );
 
                 }
                 else
                 {
                     if ((size_t)col1 >= pos)
                     {
                         // ...or remove the attribute
-                        m_blockSelectionTopLeft.RemoveAt(n);
-                        m_blockSelectionBottomRight.RemoveAt(n);
+                        m_selection.erase(m_selection.begin() + n);
                         n--;
                         count--;
                     }
                     else
-                        coords2.SetCol(pos);
+                        block.SetRightCol(pos);
                 }
             }
         }
     }
+}
 
-    count = m_colSelection.GetCount();
-    for ( n = 0; n < count; n++ )
+bool wxGridSelection::EditCurrentBlock(int startRow, int startCol,
```

Comment:
It's still a bit difficult to understand what does this function do, but I don't think "Edit" is a good name. Perhaps it should be "Extend"? Or "ExtendOrReplace"?

In fact, the fact that this function does something apparently complicated makes me wonder if we don't need 2 functions instead, one for extending the current block and another for starting a new one. Could this be simpler?